acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
-		h.AssertNil(t, err)
-		packVer, err = packVersion(packPath)
-		h.AssertNil(t, err)
-		packSemver = semver.MustParse(strings.TrimPrefix(strings.Split(packVer, " ")[0], "v"))
+		pack = managers.NewPackExecutor(t, assetManager, combo.Pack, registry, assert)
+		createBuilderPack = managers.NewPackExecutor(t, assetManager, combo.PackCreateBuilder, registry, assert)
+
+		containerManager = managers.NewTestContainerManager(t, assert, dockerCli)
+
+		stackManager = managers.NewStackManager(t, dockerCli, registry, assert, containerManager)
+		suiteManager.RegisterCleanUp("cleanup-stack-images", stackManager.Cleanup)
+		testLifecycle = managers.NewTestLifecycle(assetManager, combo.Lifecycle)
+
+		packageManager = managers.NewPackageManager(t, assert, suiteManager, createBuilderPack)
+		suiteManager.RegisterCleanUp("cleanup-package-manager", packageManager.Cleanup)
+
+		builderManager = managers.NewBuilderManager(
+			t,
+			assert,
+			registry,
+			dockerCli,
+			createBuilderPack,
+			testLifecycle,
+			packageManager,
+			stackManager,
+			combo,
+		)
+
+		buildpackManager = managers.NewBuildpackManager(t, assert)
+
+		suiteManager.RegisterCleanUp(
+			fmt.Sprintf("cleanup-builder-%s", builderManager.BuilderComboDescription()),
+			builderManager.EnsureComboBuilderCleanedUp,
+		)
 	})
 
 	it.After(func() {
-		h.AssertNil(t, os.RemoveAll(packHome))
+		pack.Cleanup()
+	})
+
+	when("create-builder invoked with invalid builder.toml", func() {
+		it("displays an error", func() {
+			h.SkipIf(t,
+				!createBuilderPack.SupportsFeature(components.BuilderTomlValidation),
+				"builder.toml validation not supported",
+			)
+
+			builderConfigPath := createBuilderPack.FixtureMustExist("invalid_builder.toml")
+
+			output, err := createBuilderPack.RunWithCombinedOutput(
+				"create-builder", "some-builder:build",
+				"--builder-config", builderConfigPath,
+			)
+			assert.NotNil(err)
+			assert.Contains(output, "invalid builder toml")
+		})
+	})
+
+	when("creating a windows builder", func() {
+		var tmpDir string
+
+		it.Before(func() {
+			h.SkipIf(t, containerManager.HostOS() != "windows", "The current Docker daemon does not support Windows-based containers")
+
+			var err error
+			tmpDir, err = ioutil.TempDir("", "create-test-builder")
+			assert.Nil(err)
+		})
+
+		it.After(func() {
+			os.RemoveAll(tmpDir)
+		})
+
+		when("experimental is disabled", func() {
+			it("fails", func() {
+				stackManager.EnsureDefaultStackCreated()
+
+				buildpack := components.NoOpBuildpack
+				buildpackManager.PlaceBuildpacksInDir(tmpDir, buildpack)
+				builderConfigPath := builderManager.NewDynamicBuilderConfig(buildpack).ConfigFile(tmpDir)
+
+				output, err := createBuilderPack.RunWithCombinedOutput(
+					"create-builder",
+					"--no-color",
+					registry.RepoName(fmt.Sprintf("test/builder-%s", h.RandString(8))),
+					"-b", builderConfigPath,
+				)
+
+				assert.NotNil(err)
+				assert.NewOutputAssertionManager(output).ReportsWindowsContainersExperimental()
```

Comment:
Can you please explain why we need this function? Why can't we do `assert.Contains(output, "Windows containers support is currently experimental")`?

Reply:
It's not so much about the testObject. `assert` has its own testObject, and `assert.Contains` also calls `Helper()`.

It's more of philosophical choice to separate the concerns of *the message exists* and *the message is exactly this*. From the perspective of our tests, I think the important thing is that the message is shown, not exactly what it is. The OutputAssertionManager becomes the single source of truth for what the message should be.

Practically it means fewer changes if we ever decide to change the wording in our output.

Reply:
Thank you @simonjjones but I didn't fully understand.
What is the difference between this case and line 479 (`assert.Contains(output, "invalid builder toml")`)?
And why does it mean fewer changes if we ever decide to change the wording in the output? It looks like it'll be either make the changes in the `it` block or in the `ReportsWindowsContainersExperimental` function.
I understand that I'm missing something but I'm not sure what is it.

Reply:
I don't think you've missed anything @yaelharel. There's no real difference between this and `assert.Contains(output, "invalid builder toml")`.  I do think both approaches are around about as clear, one small difference that may have prompted the difference is the line length of the message.

I didn't consider 100% consistency to actually be a goal. Even if it were achieved in this PR it would be hard to enforce going forward. Both approaches have validity, and people will have different preferences going forward.

> And why does it mean fewer changes if we ever decide to change the wording in the output?

This applies if the message is reused in multiple tests. That isn't the case with this particular message and probably never will be. A lot of the other messages in `OutputAssertionManager` are used repeatedly. Taking `IncludesGoogleBuilder` as an example, this output crops up on a number of occasions - as part of `suggested-builders`, `build` without a configured builder and `list-trusted-builders`. If Google decide to change the name of their builder those three tests don't really care what it changes to, but they will likely care that we're still showing the Google builder whatever it's called.

Reply:
Overall, I'm a fan of the use of the `OutputAssertionManager`. I feel like it really helped equalize the level of abstraction we were dealing with in the test; specific wording may be important to codify, but it was nice to see it pulled out of the tests themselves.

Reply:
@simonjjones, thanks again for the explanation!
I think it's worth being consistent, at least in this commit, since it'll increase the chances that the next PR will keep the same approach.
Maybe it's only me, but as you can understand from my comments, it confused me to see the two approaches in a commit that its purpose is to organize the code.